cmd/flightctl-api/main.go

```diff
@@ -95,41 +96,34 @@ func main() {
 
 	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGHUP, syscall.SIGTERM, syscall.SIGQUIT)
 
-	grpcServer := agentserver.NewAgentGrpcServer(log, cfg, grpcTlsConfig)
+	// create the agent service listener as tcp (combined HTTP+gRPC)
+	agentListener, err := net.Listen("tcp", cfg.Service.AgentEndpointAddress)
```

Comment:
without a bare tcp listener, the gRPC endpoint is unable to properly multiplex over HTTP.

We get PRI requests, which are an indication of the protocol being negotiated as HTTP1 before it hits grpc. I tried many combinations.